lib/cn/cn.c

```diff
@@ -750,21 +750,28 @@ cn_ingestv(
         err = cn_ingest_prep(cn[i], mbv[i], txid, &context, &kvsetv[i]);
         if (ev(err))
             goto done;
-        check++;
+
+        if (!kvsetv[i])
+            count--;
+        else
+            check++;
     }
+
     assert(check == count);
 
+    if (count == 0)
+        goto done; // What about cndb_txn_start???
+
```

Comment:
@gsramdasi do we need to abort the cndb txn we started at line 737?

Reply:
Ah, but that only naks if  (err != 0). which if (count == 0) err is not zero...  So maybe if I change that to be:

```
if ((err || count == 0) && txid && nak())
```

I don't think I want to set err if (count == 0)...

Or could we move the txn_start call to after the (count == 0) check???

Reply:
Nah, can't do that cuz cn_ingest_prep() commits mblocks...

Reply:
FWIW, I cannot let the ack proceed because cndb goes belly up on restart if we commit a txn with no kvsets.  It's a rats nest I don't want to untangle at the moment, which is why I just want to exit this function with minimal damage..